.github/workflows/clang-tidy.yml

```diff
@@ -32,36 +51,87 @@ concurrency:
           sudo apt-get update
-          sudo apt-get install -y clang libomp-dev ocl-icd-libopencl1 ocl-icd-opencl-dev
+          sudo apt-get install -y clang libomp-dev ocl-icd-libopencl1 ocl-icd-opencl-dev clang-tidy
+
+      # This needs to run on all platforms even if it is not used.
+      - name: Read ci.json versions file
+        id: get-versions
+        run: |
+          content=`cat ${{ github.workspace }}/oneDNN/.github/automation/aarch64/ci.json`
+          content="${content//[$'\t\r\n$ ']}"
+          echo "output=$content" >> $GITHUB_OUTPUT
+
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Clone ACL
+        run: ${{ github.workspace }}/oneDNN/.github/automation/aarch64/build_acl.sh
+        env:
+          ACL_ACTION: clone
+          ACL_ROOT_DIR: ${{ github.workspace }}/ComputeLibrary
+          ACL_VERSION: ${{ fromJson(steps.get-versions.outputs.output).dependencies.acl }}
 
-      - name: Configure oneDNN
+      # We can avoid having to actually build ComputeLibrary by faking the files
+      # that cmake/FindACL.cmake and cmake/ACL.cmake expect to see.
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Fake ACL Build
+        working-directory: ${{ github.workspace }}/ComputeLibrary/
+        env:
+          ACL_VERSION: ${{ fromJson(steps.get-versions.outputs.output).dependencies.acl }}
+        run: |
+          mkdir ./build
+          echo "arm_compute_version=${ACL_VERSION}" > ./build/arm_compute_version.embed
+          touch ./build/libarm_compute.so
+          touch ./build/libarm_compute_graph.so
+
+      - if: ${{ matrix.config.arch == 'x64' }}
+        name: Configure oneDNN - x64
+        working-directory: ${{github.workspace}}/oneDNN
         run: .github/automation/x64/build_linters.sh
         env:
           ONEDNN_ACTION: configure
 
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Configure oneDNN - AArch64
+        working-directory: ${{github.workspace}}/oneDNN
+        run: .github/automation/aarch64/build.sh
+        env:
+          BUILD_TOOLSET: clang
+          ACL_ROOT_DIR: ${{github.workspace}}/ComputeLibrary
+          ONEDNN_ACTION: configure
+
       - name: Check source files
+        working-directory: ${{github.workspace}}/oneDNN
+        shell: bash
         run: |
+          set +e
           echo -e "Checking Clang-Tidy $(clang-tidy --version)\n"
           touch clang-tidy.log
-          for file in $(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.cpp$');
-          do
-            if grep -q "$file" "build/compile_commands.json"; then
-              echo -e "\nAnalyzing $file"
-              clang-tidy -p build $file 2>&1 | tee -a clang-tidy.log
-            else
-              echo "Skipped $file as it's not built in x64 OpenMP/OpenCL configuration."
-            fi
-          done
+          analysis_list="$(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.[ch](pp|xx)?$')"
```

Comment:
clang-tidy is not designed to handle header files directly. These should be checked only in the context of cpp files that include them.